fetcher/github.go

```diff
@@ -149,3 +152,114 @@ func createClients() {
 		githubClientList = append(githubClientList, github.NewClient(tc))
 	}
 }
+
+func addEventCount(event github.Event) map[string]interface{} {
+	result := make(map[string]interface{})
+
+	// All the count metric is for PR
+	repo, devDay, devYear, additions, deletions := 0, 0, 0, 0, 0
+	// Due to we just calculate the PR event, so the pr metric should always be add 1
+	prSign, openSign, mergeSign, closeSign := 1, 0, 0, 0
+	language, devID, repoID := "", int64(0), int64(0)
+
+	// Parse event payload
+
+	// Just calculate the PR event
+	if event.Type == nil || *event.Type != "PullRequestEvent" {
+		return result
+	}
+
+	payload, err := event.ParsePayload()
+	if err != nil {
+		logger.Error("parse payload error", zap.Error(err))
+		return result
+	}
+
+	prEvent, ok := payload.(*github.PullRequestEvent)
+	if !ok || prEvent == nil {
+		logger.Error("pr payload not PullRequestEvent type")
+		return result
+	}
+
+	if prEvent.Action != nil {
+		switch *prEvent.Action {
+		case "closed":
+			if prEvent.PullRequest != nil && prEvent.PullRequest.Merged != nil && *prEvent.PullRequest.Merged {
+				mergeSign = 1
+
+				if prEvent.PullRequest.Additions != nil && prEvent.PullRequest.Deletions != nil {
+					additions = *prEvent.PullRequest.Additions
+					deletions = *prEvent.PullRequest.Deletions
+				}
+			} else {
+				closeSign = 1
+			}
+		case "opened":
+			openSign = 1
+		}
+	}
+
+	if event.Actor != nil && event.Actor.ID != nil {
+		devID = *event.Actor.ID
+	}
+
+	if prEvent.PullRequest != nil && prEvent.PullRequest.Base != nil &&
+		prEvent.PullRequest.Base.GetRepo() != nil {
+		if prEvent.PullRequest.Base.GetRepo().Language != nil &&
+			len(*prEvent.PullRequest.Base.GetRepo().Language) != 0 {
+			language = *prEvent.PullRequest.Base.GetRepo().Language
+		}
+
+		if prEvent.PullRequest.Base.GetRepo().ID != nil {
+			repoID = *prEvent.PullRequest.Base.GetRepo().ID
+		}
+	}
+
+	// Database operations
+	err = risingwave.InsertMessage(*event.ID, devID, repoID, *event.CreatedAt, language)
```

Comment:
Consider using builk insert if this is run frequently.

Reply:
But we do have some operations after it.
It might not cut down a lot of requests and bring us more legacy.

FYI, the code after `InsertMessage`:

https://github.com/pingcap-inc/ossinsight-plugin/blob/e5d98e831f531c4a91577b1c380871d3be34c2fc/fetcher/github.go#L224-L241

Reply:
that's reasonable, thanks for the explanation!